Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM mcr.microsoft.com/dotnet/sdk:6.0 AS build-env
+WORKDIR /source
+
+# Copy all .csproj files and use `dotnet restore` to install dependencies.
+# Doing this first will allow the dependencies to stay cached if the csproj
+# files didn't change, even if other code did.
+COPY *.sln .
+COPY **/*.csproj ./
+RUN dotnet sln list \
+    | tail -n +3 \
+    | xargs -I {} sh -c \
+        'target="{}"; dir="${target%/*}"; file="${target##*/}"; mkdir -p -- "$dir"; mv -- "$file" "$target"'
```

Comment:
Taken from https://stackoverflow.com/a/65390923, this is a hacky thing you need to do to move each of the .csproj files to the proper directory in the docker image. A little annoying but better than hardcoding it because if anything about the project structure is changed (new project added, project renamed, etc) it won't require any changes to the Dockerfile